### DISCUSSION TITLE
Possible Model Improvement

### DISCUSSION THREAD

**Discussion by ptth222** [ISA-tools/isa-api#533] [Ideas]
I was looking at the Investigation class in isatools/model/investigation.py and I noticed an effort to enforce element types in some lists. For example, the ontology_source_references setter looks to see that all the elements of the list are of type OntologySource. I was playing around with this and you can still append a non-OntologySource to the ontology_source_references. If you want to be more robust about it I found a generalized solution for a typed list here: https://gist.github.com/KrzysztofCiba/4579691. You could set the ontology_source_references to be one of these typed lists and not need specialized setters or methods to type check. This would work for other list properties you want to enforce type on as well, such as studies.

**Comment by proccaserra** [comment]
@ptth222 we are merging a number of changes that will address the issue with Ontology Source References.
